File: app/src/main/java/com/lomza/moviesroom/db/MovieDao.java
package com.lomza.moviesroom.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

/**
 * @author Antonina
 */
@Dao
public interface MovieDao {
    @Query("SELECT * FROM movie WHERE title = :title LIMIT 1")
    Movie findMovieByTitle(String title);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Movie... directors);

    @Update(onConflict = OnConflictStrategy.IGNORE)
    void update(Movie director);

    @Query("DELETE FROM movie")
    void deleteAll();

    @Query("SELECT * FROM movie ORDER BY title ASC")
    LiveData<List<Movie>> getAllMovies();
}

File: app/src/main/java/com/lomza/moviesroom/db/MovieDao.kt
package com.lomza.moviesroom.db

import androidx.lifecycle.LiveData
import androidx.room.*

/**
 * @author Antonina
 */
@Dao
interface MovieDao {

    @Query("SELECT * FROM movie WHERE title = :title LIMIT 1")
    suspend fun findMovieByTitle(title: String?): Movie?

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(vararg directors: Movie)

    @Update(onConflict = OnConflictStrategy.IGNORE)
    suspend fun update(director: Movie)

    @Query("DELETE FROM movie")
    suspend fun deleteAll()

    @get:Query("SELECT * FROM movie ORDER BY title ASC")
    val allMovies: LiveData<List<Movie>>
}